src/parser/gjs-gts-parser.js

```diff
@@ -40,26 +45,27 @@ module.exports = {
     }
 
     try {
-      result = isTypescript
-        ? typescriptParser.parseForESLint(jsCode, {
-            ...options,
-            ranges: true,
-            extraFileExtensions: ['.gts', '.gjs'],
-            filePath,
-          })
-        : babelParser.parseForESLint(jsCode, {
-            ...options,
-            requireConfigFile: false,
-            ranges: true,
-          });
+      result =
+        isTypescript || useTypescript
```

Comment:
I think we don’t need a `isTypescript` check anymore: if you really want to use Babel to parse your typescript (which you can do with the right plugins; the reason maybe that you are adopting experimental syntax), you should be allowed to override it. What do you think?

Reply:
Is it compatible with typescript eslint rules?

Reply:
Good question. But I think it’s irrelevant though. You are in control of your whole setup, and here you would be going out of your way to opt in to Babel, so presumably you know what you are doing and you are picking the rules that works with your parser. We are basically giving you (retaining) the same amount of configurability as eslint itself gives you, whereas if this is dynamically picked per file like this, you can’t do anything about it if you don’t like what we picked (you can always have overrides for specific files with different parser options, too)

We are in the outer edge of the edge cases though. Not that I have an actual use case for it, just mostly thinking about what seems necessary/to be a good configurable api here

Reply:
It mostly doesn't matter since the optional defaults to true. When it _does_ matter, it probably looks something like this:

```js
{
  // set globally and applies to all files
  parser: 'ember-eslint-parser',
  parserOption: {
    useBabel: true,
    babelOptions: { ... }
  }
}
```

So it comes down to what you _think_ that means. Personally, I see that as the equivalent of:

```js
{
  // set globally and applies to all files
  parser: '@babel/eslint-parser+gjs-gts-support',
  parserOption: {
    useBabel: true,
    babelOptions: { ... }
  }
}
```

Where as, it currently means something kind of weird (ignore these options if the file happens to be TypeScript).

If you want the current behavior, I would expect something like this:

```js
{
  // set globally and applies to all files
  parser: 'ember-eslint-parser',
  overrides: [
    {
      files: ["*/*.js", "*/*.gjs"],
      parserOption: {
        useBabel: true,
        babelOptions: { ... }
      }
    },
    {
      files: ["*/*.ts", "*/*.gts"],
      parserOption: {
        // not needed, just to be explicit
        useBabel: false,
      }
    },
  ]
}
```

which seems clearer to me (and mostly I am vary of some kind of "un-overridable auto-detection")